File: app/src/main/java/com/example/colorful_android/Home/NaviActivity.java
//package com.example.colorful_android.Home;
//
//import android.os.Bundle;
//import android.view.MenuItem;
//
//import androidx.annotation.NonNull;
//import androidx.appcompat.app.AppCompatActivity;
//
//import com.example.colorful_android.Fragment.ColorFragment;
//import com.example.colorful_android.Fragment.HomeFragment;
//import com.example.colorful_android.Fragment.MypageFragment;
//import com.example.colorful_android.Fragment.NaviActivity;
//import com.example.colorful_android.Fragment.SearchFragment;
//import com.example.colorful_android.R;
//import com.example.colorful_android.databinding.ActivityNaviBinding;
//import com.google.android.material.bottomnavigation.BottomNavigationView;
//
//
//public class NaviActivity extends AppCompatActivity {
//    private BottomNavigationView bottomNavigationView;
//    private ActivityNaviBinding binding;
//
//    @Override
//    protected void onCreate(Bundle savedInstanceState) {
//        super.onCreate(savedInstanceState);
//
//        setContentView(R.layout.activity_main);
//        bottomNavigationView = findViewById(R.id.bottomNavigationView);
//
//
//        HomeFragment homeFragment = new HomeFragment();
//
//        MypageFragment mypageFragment = new MypageFragment();
//        ColorFragment colorFragment = new ColorFragment();
//        SearchFragment searchFragment = new SearchFragment();
//
//        getSupportFragmentManager().beginTransaction().replace(R.id.main_layout,homeFragment).commitAllowingStateLoss();
//
////        binding = ActivityNaviBinding.inflate(layoutInflater)
////        val view = binding.root
////        setContentView(view)
////
////        //맨 처음 보여줄 프래그먼트 설정
////        setFragment(TAG_HOME, HomeFragment())
//
//        bottomNavigationView.setOnNavigationItemSelectedListener(new BottomNavigationView.OnNavigationItemSelectedListener() {
//
//            @Override
//            public boolean onNavigationItemSelected(@NonNull MenuItem menuItem) {
//                switch (menuItem.getItemId()){
//                    //menu_bottom.xml에서 지정해줬던 아이디 값을 받아와서 각 아이디값마다 다른 이벤트를 발생시킵니다.
//                    case R.id.tab1:{
//                        getSupportFragmentManager().beginTransaction() .replace(R.id.main_layout,fragment_home).commitAllowingStateLoss();
//                        return true;
//                    }
//
//
//                    case R.id.tab2:{
//                        getSupportFragmentManager().beginTransaction()
//                                .replace(R.id.main_layout,fragment2).commitAllowingStateLoss();
//                        return true;
//                    }
//
//
//                    case R.id.tab3:{
//                        getSupportFragmentManager().beginTransaction()
//                                .replace(R.id.main_layout,fragment3).commitAllowingStateLoss();
//                        return true;
//                    }
//
//
//
//                    default: return false;
//
//
//                }
//    }
//
//
//
////        setFragment(TAG_HOME, HomeFragment())
////
////        binding.navigationView.setOnItemSelectedListener { item ->
////        when(item.itemId) {
////        R.id.calenderFragment -> setFragment(TAG_CALENDER, CalenderFragment())
////        R.id.homeFragment -> setFragment(TAG_HOME, HomeFragment())
////        R.id.myPageFragment-> setFragment(TAG_MY_PAGE, MyPageFragment())
////        }
////        true
////        }
//}

File: app/src/main/java/com/example/colorful_android/NaviActivity.kt
package com.example.colorful_android

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import com.example.colorful_android.databinding.ActivityNaviBinding

private const val TAG_HOME = "home_fragment"
private const val TAG_SEARCH = "search_fragment"
private const val TAG_COLOR = "color_fragment"
private const val TAG_MYPAGE = "mypage_fragment"

class NaviActivity : AppCompatActivity() {

    private lateinit var binding :ActivityNaviBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityNaviBinding.inflate(layoutInflater)
        setContentView(binding.root)

        setFragment(TAG_HOME, HomeFragment())

        binding.navMain.run {
            setOnItemSelectedListener { item ->
                when(item.itemId){
                    R.id.homeFragment -> setFragment(TAG_HOME, HomeFragment())
                    R.id.searchFragment -> setFragment(TAG_SEARCH, SearchFragment())
                    R.id.colorFragment -> setFragment(TAG_COLOR, ColorFragment())
                    R.id.mypageFragment-> setFragment(TAG_MYPAGE, MypageFragment())
                }
                true
            }
        }

    }

    private fun setFragment(tag: String, fragment: Fragment) {
        val manager : FragmentManager = supportFragmentManager
        val fragTransaction = manager.beginTransaction()

        if(manager.findFragmentByTag(tag) == null) {
            fragTransaction.add(R.id.mainFramLayout, fragment, tag)
        }

        val home = manager.findFragmentByTag(TAG_HOME)
        val search = manager.findFragmentByTag(TAG_SEARCH)
        val color = manager.findFragmentByTag(TAG_COLOR)
        val mypage = manager.findFragmentByTag(TAG_MYPAGE)

        if(home != null){
            fragTransaction.hide(home)
        }

        if(search != null){
            fragTransaction.hide(search)
        }

        if(color != null){
            fragTransaction.hide(color)
        }

        if(mypage != null){
            fragTransaction.hide(mypage)
        }

        if(tag == TAG_HOME){
            if(home != null){
                fragTransaction.show(home)
            }
        }
        else if(tag == TAG_SEARCH){
            if(search != null){
                fragTransaction.show(search)
            }
        }
        else if(tag == TAG_COLOR){
            if(color != null){
                fragTransaction.show(color)
            }
        }
        else if(tag == TAG_MYPAGE){
            if(mypage != null){
                fragTransaction.show(mypage)
            }
        }

        fragTransaction.commitAllowingStateLoss()

    }
}